azurerm/internal/services/appplatform/resource_arm_spring_cloud_service.go

```diff
@@ -0,0 +1,591 @@
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+
+	v := input[0].(map[string]interface{})
+	if v == nil {
+		return nil
+	}
+
+	result := appplatform.ConfigServerGitProperty{
+		URI: utils.String(v["uri"].(string)),
+	}
+
+	if label := v["label"].(string); label != "" {
+		result.Label = utils.String(label)
+	}
+	if searchPaths := v["search_paths"].([]interface{}); len(searchPaths) > 0 {
+		result.SearchPaths = utils.ExpandStringSlice(searchPaths)
+	}
+	if httpBasicAuth := v["http_basic_auth"].([]interface{}); len(httpBasicAuth) > 0 {
+		v := httpBasicAuth[0].(map[string]interface{})
+		result.Username = utils.String(v["username"].(string))
+		result.Password = utils.String(v["password"].(string))
+	}
+	if sshAuth := v["ssh_auth"].([]interface{}); len(sshAuth) > 0 {
+		v := sshAuth[0].(map[string]interface{})
+		result.PrivateKey = utils.String(v["private_key"].(string))
+		result.StrictHostKeyChecking = utils.Bool(v["strict_host_key_checking_enabled"].(bool))
+
+		if hostKey := v["host_key"].(string); hostKey != "" {
+			result.HostKey = utils.String(hostKey)
+		}
+		if hostKeyAlgorithm := v["host_key_algorithm"].(string); hostKeyAlgorithm != "" {
+			result.HostKeyAlgorithm = utils.String(hostKeyAlgorithm)
+		}
+	}
+
+	if warnings, ok := v["repository"]; ok {
+		result.Repositories = expandArmSpringCloudGitPatternRepository(warnings.([]interface{}))
+	}
+
+	return &result
+}
+
+func expandArmSpringCloudGitPatternRepository(input []interface{}) *[]appplatform.GitPatternRepository {
+	results := make([]appplatform.GitPatternRepository, 0)
+	for _, item := range input {
+		v := item.(map[string]interface{})
+
+		result := appplatform.GitPatternRepository{
+			Name: utils.String(v["name"].(string)),
+			URI:  utils.String(v["uri"].(string)),
+		}
+
+		if label := v["label"].(string); len(label) > 0 {
+			result.Label = utils.String(label)
+		}
+		if pattern := v["pattern"].([]interface{}); len(pattern) > 0 {
+			result.Pattern = utils.ExpandStringSlice(pattern)
+		}
+		if searchPaths := v["search_paths"].([]interface{}); len(searchPaths) > 0 {
+			result.SearchPaths = utils.ExpandStringSlice(searchPaths)
+		}
+		if httpBasicAuth := v["http_basic_auth"].([]interface{}); len(httpBasicAuth) > 0 {
+			v := httpBasicAuth[0].(map[string]interface{})
+			result.Username = utils.String(v["username"].(string))
+			result.Password = utils.String(v["password"].(string))
+		}
+		if sshAuth := v["ssh_auth"].([]interface{}); len(sshAuth) > 0 {
+			v := sshAuth[0].(map[string]interface{})
+			result.PrivateKey = utils.String(v["private_key"].(string))
+			result.StrictHostKeyChecking = utils.Bool(v["strict_host_key_checking_enabled"].(bool))
+
+			if hostKey := v["host_key"].(string); hostKey != "" {
+				result.HostKey = utils.String(hostKey)
+			}
+			if hostKeyAlgorithm := v["host_key_algorithm"].(string); hostKeyAlgorithm != "" {
+				result.HostKeyAlgorithm = utils.String(hostKeyAlgorithm)
+			}
+		}
```

Comment:
as with the top level repo, we should test against these both being set and error out to the user to prevent both being attempted to be sent to the API since only one will be accepted (based on your comments from the service team that this is conditional based on the protocol used in the URI).

Reply:
@jackofallops I have added check to prevent setting both the fields, please have a look. Thanks for your suggestion